src/main/java/com/aws/greengrass/lifecyclemanager/GreengrassService.java

```diff
@@ -449,6 +449,7 @@ protected CompletableFuture<Void> close(boolean waitForDependers) {
                         logger.error("Interrupted waiting for dependers to exit");
                     }
                 }
+                externalDependenciesTopic.remove();
```

Comment:
As communicated offline, directly calling `externalDependenciesTopic.remove()` will remove the entire node. You should just remove the watcher